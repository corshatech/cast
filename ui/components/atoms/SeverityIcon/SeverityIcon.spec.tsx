import { render } from '@testing-library/react';

import type { Severity } from '@/lib/findings';
import { SeverityIcon } from './';

describe('SeverityIcon', () => {
  test.each(['none', 'low', 'medium', 'high', 'critical', 'UNKNOWN_-InVALID'])(
    'matches snapshot for severity %p',
    (severity) => {
      const { asFragment } = render(<SeverityIcon severity={severity as Severity}/>)
      expect(asFragment()).toMatchSnapshot()
    })
});
