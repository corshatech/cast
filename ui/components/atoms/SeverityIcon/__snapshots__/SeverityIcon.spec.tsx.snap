// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeverityIcon matches snapshot for severity "UNKNOWN_-InVALID" 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
    data-testid="LightbulbIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M9 21c0 .5.4 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19C8.1 2 5 5.1 5 9c0 2.4 1.2 4.5 3 5.7V17c0 .5.4 1 1 1h6c.6 0 1-.5 1-1v-2.3c1.8-1.3 3-3.4 3-5.7 0-3.9-3.1-7-7-7z"
    />
  </svg>
</DocumentFragment>
`;

exports[`SeverityIcon matches snapshot for severity "critical" 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
    data-testid="CrisisAlertIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M14.5 2.5c0 1.5-1.5 6-1.5 6h-2S9.5 4 9.5 2.5C9.5 1.12 10.62 0 12 0s2.5 1.12 2.5 2.5zM12 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm4.08-4.89c.18-.75.33-1.47.39-2.06C19.75 4.69 22 8.08 22 12c0 5.52-4.48 10-10 10S2 17.52 2 12c0-3.92 2.25-7.31 5.53-8.95.07.59.21 1.32.39 2.06C5.58 6.51 4 9.07 4 12c0 4.42 3.58 8 8 8s8-3.58 8-8c0-2.93-1.58-5.49-3.92-6.89zM18 12c0 3.31-2.69 6-6 6s-6-2.69-6-6c0-2 .98-3.77 2.48-4.86.23.81.65 2.07.65 2.07C8.43 9.93 8 10.92 8 12c0 2.21 1.79 4 4 4s4-1.79 4-4c0-1.08-.43-2.07-1.13-2.79 0 0 .41-1.22.65-2.07C17.02 8.23 18 10 18 12z"
    />
  </svg>
</DocumentFragment>
`;

exports[`SeverityIcon matches snapshot for severity "high" 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
    data-testid="ErrorIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
    />
  </svg>
</DocumentFragment>
`;

exports[`SeverityIcon matches snapshot for severity "low" 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-3adcp6-MuiSvgIcon-root"
    data-testid="ExpandCircleDownIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
    />
  </svg>
</DocumentFragment>
`;

exports[`SeverityIcon matches snapshot for severity "medium" 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-3adcp6-MuiSvgIcon-root"
    data-testid="WarningIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
    />
  </svg>
</DocumentFragment>
`;

exports[`SeverityIcon matches snapshot for severity "none" 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
    data-testid="LightbulbIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M9 21c0 .5.4 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19C8.1 2 5 5.1 5 9c0 2.4 1.2 4.5 3 5.7V17c0 .5.4 1 1 1h6c.6 0 1-.5 1-1v-2.3c1.8-1.3 3-3.4 3-5.7 0-3.9-3.1-7-7-7z"
    />
  </svg>
</DocumentFragment>
`;
